dockerclient.go

```diff
@@ -231,6 +231,48 @@ func (client *DockerClient) ContainerChanges(id string) ([]*ContainerChanges, er
 	return changes, nil
 }
 
+func (client *DockerClient) readJSONStream(stream io.ReadCloser, decode func(*json.Decoder) decodingResult, stopChan <-chan struct{}) <-chan decodingResult {
+	resultChan := make(chan decodingResult)
+	internalResultsChan := make(chan decodingResult)
+	stillListening := make(chan struct{})
+
+	go func() {
```

Comment:
good call, you're very right. this is way better. I had the separate goroutine because of a slightly different approach I was taking with the `closeChan` which which required it, but there's no reason at all to have a separate goroutine now. thanks for explaining :)